lib/matplotlib/table.py

```diff
@@ -839,4 +839,7 @@ def table(ax,
             table.auto_set_column_width(-1)
 
     ax.add_table(table)
+    # Set the fontsize if provided
+    if fontsize is not None:
+        table.set_fontsize(fontsize)
```

Comment:
I suggest putting this before `add_table` because it is part of defining the table so belongs with the cell adding logic.  Also it would be worth expanding the comment to explain _why_ we do this here rather than leaving it as part of kwargs.  Something like

`# set_fontsize is only effective after cells are added`